src/Explorer/QueryCopilot/QueryCopilotTab.tsx

```diff
@@ -230,6 +234,23 @@ export const QueryCopilotTab: React.FC<QueryCopilotTabProps> = ({
     useCommandBar.getState().setContextButtons(getCommandbarButtons());
   }, [query, selectedQuery]);
 
+  React.useEffect(() => {
+    if (initialInput) {
+      generateSQLQuery();
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (!inputEdited.current) {
+      setTimeout(() => {
+        if (!inputEdited.current) {
+          toggleCopilotTeachingBubbleVisible();
+          inputEdited.current = true;
+        }
+      }, 30000);
+    }
+  });
```

Comment:
I think we only need run this useEffect once the first time user opens the tab, so you should be able to combine this with the useEffect defined above on line 237 (with a dependency array of []).